### PR TITLE
uwsgi: work around bug with case-sensitive HFS

### LINKED ISSUE #45560: uwsgi failed to build on 10.11.1
uwsgi itself is building ok, but overall installation fails:

```
➜  ~ brew install uwsgi --with-python3 --verbose

...

############## end of uWSGI configuration #############
total build time: 22 seconds
*** uWSGI is ready, launch it with ./uwsgi ***

...

==> python uwsgiconfig.py --plugin plugins/alarm_speech brew
ld: framework not found appkit
clang: error: linker command failed with exit code 1 (use -v to see invocation)
using profile: buildconf/brew.ini
detected include path: ['/usr/local/opt/openssl/include', '/usr/local/opt/readline/include', '/usr/local/opt/sqlite/include', '/usr/local/include', '/usr/include/libxml2', '/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers', '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.0/include', '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include', '/usr/include', '/System/Library/Frameworks', '/Library/Frameworks']
*** uWSGI building and linking plugin plugins/alarm_speech ***
[clang] /usr/local/Cellar/uwsgi/2.0.11.2/libexec/uwsgi/alarm_speech_plugin.so
*** unable to build alarm_speech plugin ***
==> Formula
Path: /usr/local/Library/Formula/uwsgi.rb
==> Configuration
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Homebrew/homebrew
HEAD: daf7b96b8fb064c359910fbe83abe95a6e8dd828
Last commit: 3 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit ivybridge
OS X: 10.11.1-x86_64
Xcode: 7.1
CLT: 7.1.0.0.1.1444952191
Clang: 7.0 build 700
X11: N/A
System Ruby: 2.0.0-p645
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_66
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
CFLAGS: -I/usr/local/opt/openssl/include -arch x86_64
LDFLAGS: -L/usr/local/opt/openssl/lib -arch x86_64
MAKEFLAGS: -j4
CMAKE_PREFIX_PATH: /usr/local/opt/openssl:/usr/local/opt/readline:/usr/local/opt/sqlite:/usr/local
CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
CMAKE_FRAMEWORK_PATH: /usr/local/opt/python3/Frameworks
PKG_CONFIG_PATH: /usr/local/opt/openssl/lib/pkgconfig:/usr/local/opt/pcre/lib/pkgconfig:/usr/local/opt/sqlite/lib/pkgconfig:/usr/local/opt/xz/lib/pkgconfig:/usr/local/opt/python3/lib/pkgconfig:/usr/local/opt/yajl/share/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.11
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Library/ENV/4.3:/usr/local/opt/pkg-config/bin:/usr/local/opt/openssl/bin:/usr/local/opt/pcre/bin:/usr/local/opt/yajl/bin:/usr/local/opt/sqlite/bin:/usr/local/opt/gdbm/bin:/usr/local/opt/xz/bin:/usr/local/opt/python3/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: uwsgi 2.0.11.2 did not build
➜  ~  brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: The filesystem on / appears to be case-sensitive.
The default OS X filesystem is case-insensitive. Please report any apparent problems.

```

### PR DESCRIPTION
**Pull request by apjanke**
Fixes #45560 

Temporary workaround until the upstream unbit/uwsgi#1127 gets fixed in a future release.

### CODE REVIEW

File: Library/Formula/uwsgi.rb [hunk 1]
```diff
@@ -81,7 +85,7 @@ def install
                "stats_pusher_socket", "symcall", "syslog",
                "transformation_chunked", "transformation_gzip",
                "transformation_offload", "transformation_tofile",
-               "transformation_toupper", "ugreen", "webdav", "zergpool",]
+               "transformation_toupper", "ugreen", "webdav", "zergpool"]
```
Comment (apjanke) [inline Library/Formula/uwsgi.rb]: Audit fix.

File: Library/Formula/uwsgi.rb [hunk 2]
```diff
@@ -45,6 +45,10 @@ class Uwsgi < Formula
   depends_on "v8" => :optional
   depends_on "zeromq" => :optional
 
+  # Fix build on case-sensitive filesystems
+  # https://github.com/Homebrew/homebrew/issues/45560
+  patch :DATA
```
Comment (DomT4) [inline Library/Formula/uwsgi.rb]: Is it worth doing this as an `inreplace` rather than a `patch` if it's just one line to change?

Comment (apjanke) [inline Library/Formula/uwsgi.rb]: Switched it to `inreplace` for better readability. Still builds on case-sensitive HFS.

**Comment by apjanke** [comment]
@BrewTestBot test this please

File: Library/Formula/uwsgi.rb [hunk 3]
```diff
@@ -45,6 +45,10 @@ class Uwsgi < Formula
   depends_on "v8" => :optional
   depends_on "zeromq" => :optional
 
+  # Fix build on case-sensitive filesystems
+  # https://github.com/Homebrew/homebrew/issues/45560
+  inreplace "plugins/alarm_speech/uwsgiplugin.py", "'-framework appkit'", "'-framework AppKit'"
```
Comment (mikemcquaid) [inline Library/Formula/uwsgi.rb]: Works for me. Can you submit upstream and add a link to that submission?

Comment (apjanke) [inline Library/Formula/uwsgi.rb]: Submitted to upstream and merged in head already. These folks don't dawdle.

Added a link to the upstream PR.

File: Library/Formula/uwsgi.rb [hunk 4]
```diff
@@ -45,6 +45,11 @@ class Uwsgi < Formula
   depends_on "v8" => :optional
   depends_on "zeromq" => :optional
 
+  # Fix build on case-sensitive filesystems
+  # https://github.com/Homebrew/homebrew/issues/45560
+  # https://github.com/unbit/uwsgi/pull/1128
+  inreplace "plugins/alarm_speech/uwsgiplugin.py", "'-framework appkit'", "'-framework AppKit'"
```
Comment (DomT4) [inline Library/Formula/uwsgi.rb]: `inreplace` is in the wrong place :). Needs to be inside `def install`.

Comment (apjanke) [inline Library/Formula/uwsgi.rb]: Oops! Fixed.

**Pull request closed without merging**